Feat/habits/dbconnector

JorgePS14 opened the pull request:
## Changes:
- Changed hardcoded values to environment variables using the Godotenv package.
- Made some minor changes to file names and variable names that were migrated from previous GuessIt code.
- Commited several changes (mostly to HabitsDb and HabitsController) which I honestly do not remember, but it all seemed to work flawlessly when running the main file, so that's a success in my books.

## How to review:
- You must create a local database "habits" using mongodb in your system for testing purposes.
- If you haven't already installed godotenv in your system, run the following command:
```shell
go get github.com/joho/godotenv
```
- Once the previous two steps have been followed, just go to the habits folder and run the main file. Output should be:
```shell
Loaded .env file successfully.
Connected to MongoDB!
```

## Notes:
- You will need the .env file to connect to the mongoDB database, so feel free to send me a private message and request the env file.

Commit:
Added connection to local database using env variables

Commit:
Changed users.go to habits.go

Commit:
Honestly don't know, but moved some things around in the habitController, and everything seems to work fine

Commit:
Changed the word user to habit in some files

LuisPalominoTrevilla reviewed (changes requested):
Me da este error:
![image](https://user-images.githubusercontent.com/22354310/57574803-69312e00-7405-11e9-855f-446f3b70599d.png)

Además, pusiste el .env visible

LuisPalominoTrevilla commented:
> Me da este error:
> ![image](https://user-images.githubusercontent.com/22354310/57574803-69312e00-7405-11e9-855f-446f3b70599d.png)
> 
> Además, pusiste el .env visible

Con este comando se resuelve:
`go clean -modcache`

- Agrega eso en el README de este microservicio para futura referencia.

- Por cierto, en el archivo go.mod tienes muchas dependencias de guessit que no se usan aquí. Quítalas, si no, se tarda mucho el correr el programa la primera vez

Commit:
Removed Guessit dependencies that shouldn't be there

JorgePS14 committed:
Deleted .env

I really do not know why this ended up here, I could've sworn I added a gitignore. Guess that'll be my next commit.

Commit:
Merge branch 'develop' of https://github.com/LuisPalominoTrevilla/TaskManagerSystemBackend into feat/habits/dbconnector

Commit:
Updated .gitignore

Commit:
Merge branch 'feat/habits/dbconnector' of https://github.com/LuisPalominoTrevilla/TaskManagerSystemBackend into feat/habits/dbconnector

JorgePS14 committed:
Update README.md

LuisPalominoTrevilla reviewed (approved):
Dale merge y continúa con todo
